File: ABC/175/b_ans.c
#include <stdio.h>
#include <string.h>

void mysort (int *x, int N);

int main(){
  // input
  int N; scanf("%d", &N);
  int i, j, k, L[N];
  for (i = 0; i < N; ++i){
    scanf("%d", &L[i]);
  }

  // compute
  mysort(L, N);
  int cnt = 0;
  for (i = 0; i < N; ++i){
    for (j = 0; j < i; ++j){
      for (k = 0; k < j; ++k){
        if (L[i]!=L[j] && L[j]!=L[k] && L[k]+L[j] > L[i]){
          cnt++;
        }
      }
    }
  }

  // output
  printf("%d\n", cnt);
  return 0;
}

void mysort (int *x, int N) {
  int i, j;
  int tmp = 0;

  // compute
  for (i = 0; i < N; ++i) {
    for (j = i; j < N; ++j) {
      if (x[i] > x[j]) {
        tmp = x[i];
        x[i] = x[j];
        x[j] = tmp;
      }
    }
  }
}

File: abc/197/b_ans.py
# ABC197B - Visibility

def main():
    # input
    H, W, X, Y = map(int, input().split())
    Ss = []
    for _ in range(H):
        Ss.append(str(input()))

    # compute
    X -= 1
    Y -= 1
    i, j, cnt = 0, 0, -3

    while X-i >= 0:
        if Ss[X-i][Y] == '.':
            cnt += 1
        else:
            break
        i += 1
    while Y-j >= 0:
        if Ss[X][Y-j] == '.':
            cnt += 1
        else:
            break
        j += 1

    i, j = 0, 0
    while X+i < H:
        if Ss[X+i][Y] == '.':
            cnt += 1
        else:
            break
        i += 1
    while Y+j < W:
        if Ss[X][Y+j] == '.':
            cnt += 1
        else:
            break
        j += 1

    # output
    print(cnt)


if __name__ == '__main__':
    main()
